base/linalg/triangular.jl

```diff
@@ -889,6 +889,196 @@ for (f, g) in ((:/, :A_rdiv_B!), (:A_rdiv_Bc, :A_rdiv_Bc!), (:A_rdiv_Bt, :A_rdiv
     end
 end
 
+# Complex matrix logarithm for the upper triangular factor, see:
+#   Al-Mohy and Higham, "Improved inverse  scaling and squaring algorithms for
+# the matrix logarithm", SIAM J. Sci. Comput., 34(4), (2012), pp. C153-C169.
+#   Al-Mohy, Higham and Relton, "Computing the Frechet derivative of the matrix
+# logarithm and estimating the condition number", SIAM J. Sci. Comput., 35(4),
+# (2013), C394-C410.
+#   http://eprints.ma.man.ac.uk/1687/02/logm.zip
+function logm{T}(A0::UpperTriangular{T})
```

Comment:
I still think you'd have to restrict this to `T<:Union{Float64,Complex{Float64}}`.